### PR TITLE
feat: Add possibilities to archive t4c instances

### LINKED ISSUE #970: Archive TeamForCapella servers
Currently, it's not possible to archive TeamForCapella servers from the Collaboration Manager. Archiving should work as follows:

- Archive a TeamForCapella server. The TeamForCapella server integration is read-only. When linking a T4C repository to a model, the server should not be selectable (marked as archives, greyed out in the dropdown menu). 
- Unarchive a TeamForCapella server. Reverts the archival of the server.

### PR DESCRIPTION
**Pull request by dominik003**
This PR adds an archive flag to the t4c instances. When set the t4c instances should be grayed out in the drop down menu (i.e., no longer be selectable) and one should no longer be able to edit the instance. However, existing linked models should still work as expected.

Resolves #970

### CODE REVIEW

File: frontend/src/app/services/settings/t4c-instance.service.ts [hunk 1]
```diff
@@ -58,6 +61,12 @@ export class T4CInstanceService {
   );
   public readonly t4cInstance$ = this._t4cInstance.asObservable();
 
+  public readonly unarchivedT4cInstances$ = this._t4cInstances.pipe(
+    map((t4cInstances) =>
+      t4cInstances?.filter((t4cInstance) => !t4cInstance.is_archived)
+    )
+  );
```
Comment (Paula-Kli) [inline frontend/src/app/services/settings/t4c-instance.service.ts:68]: why is that needed? it seems to be unused to me?

File: frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.html [hunk 2]
```diff
@@ -176,9 +176,24 @@ <h2 *ngIf="!existing">Add a Team4Capella instance</h2>
...
+          "
+          class="flex justify-between"
+        >
+          <div *ngIf="!isArchived; else archivePlaceholder">
+            <button mat-flat-button color="primary" (click)="enableEditing()">
+              Edit
+            </button>
+          </div>
+          <ng-template #archivePlaceholder>
+            <div style="flex-grow: 1"></div>
```
Comment (Paula-Kli) [inline frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.html:193]: this needs to be transferred to tailwind css, doesn't it?

File: frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.ts [hunk 3]
```diff
@@ -151,6 +154,22 @@ export class EditT4CInstanceComponent implements OnInit, OnDestroy {
...
+    if (this.instanceId) {
+      this.t4cInstanceService
+        .updateInstance(this.instanceId, {
+          is_archived: !this.isArchived,
+        })
+        .subscribe((instance) => {
+          this.isArchived = instance.is_archived;
+          this.toastService.showSuccess(
+            'Instance updated',
+            `The instance “${instance.name}” is now archived.`
```
Comment (Paula-Kli) [inline frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.ts:167]: possible that I missed something but shouldn't it be something like
```suggestion
            'Instance updated',
            `The instance “${instance.name}” is now ${this.isArchived? "archived" : "unarchived"}.`
```

**Review by Paula-Kli — CHANGES_REQUESTED** [review]
looks already good to me

**Comment by dominik003** [comment]
Add `archive` as archive icon

File: frontend/src/app/services/settings/t4c-instance.service.ts (hunk 1 above)
Comment (dominik003) [reply frontend/src/app/services/settings/t4c-instance.service.ts:68]: Good catch, deleted it.

File: frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.ts (hunk 3 above)
Comment (dominik003) [reply frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.ts:167]: Good catch!

File: backend/capellacollab/alembic/versions/f7bf9456cfc9_add_archive_flag.py
Comment (MoritzWeber0) [inline backend/capellacollab/alembic/versions/f7bf9456cfc9_add_archive_flag.py:1]: _This comment is not related to this file._

We need a bit of user documentation about the feature:
- How to archive a TeamForCapella instance
- What are the effects? What is no longer possible? What happens to existing repositories / repository links?

File: backend/capellacollab/settings/modelsources/t4c/models.py [hunk 4]
```diff
@@ -79,6 +79,8 @@ class DatabaseT4CInstance(database.Base):
         back_populates="instance", cascade="all, delete"
```
Comment (MoritzWeber0) [inline backend/capellacollab/settings/modelsources/t4c/models.py:79]: A server side validation would be good for all operations which are not permitted anymore. 

As far as I can see this only has an impact on linking T4C repositories to Capella models. We should prevent this for archives repositories.

File: frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.html [hunk 5]
```diff
@@ -176,9 +176,25 @@ <h2 *ngIf="!existing">Add a Team4Capella instance</h2>
...
+          <div *ngIf="!isArchived; else archivePlaceholder">
+            <button mat-flat-button color="primary" (click)="enableEditing()">
+              Edit
+            </button>
+          </div>
+          <ng-template #archivePlaceholder>
+            <div class="grow"></div>
+          </ng-template>
+          <button mat-flat-button color="primary" (click)="toggleArchive()">
+            <mat-icon class="mat-icon-position left">archive</mat-icon>
```
Comment (MoritzWeber0) [inline frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.html:196]: Unarchive has the wrong symbol. It also shows the archive symbol, instead it should use the `unarchive` symbol.

**Review by MoritzWeber0 — APPROVED** [review]
Looks good and works perfectly. Just three minor improvement suggestions.

File: backend/capellacollab/settings/modelsources/t4c/models.py (hunk 4 above)
Comment (dominik003) [reply backend/capellacollab/settings/modelsources/t4c/models.py:79]: I've introduced a `get_existing_unarchived_instance` injectable and implemented its use within the `create_t4c_model` route, replacing the previous utilization of the `get_existing_instance` injectable. Ideally, the new injectable could have been constructed to directly utilize `get_existing_instance`. However, since we do not inject it but rather invoke it directly within the `create_t4c_model` route, adopting such an approach would introduce additional overhead.

File: frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.html (hunk 5 above)
Comment (dominik003) [reply frontend/src/app/settings/modelsources/t4c-settings/edit-t4c-instance/edit-t4c-instance.component.html]: Done (didn't know that there is an `unarchive` symbol). Also has to be addressed in #1062

File: backend/capellacollab/alembic/versions/f7bf9456cfc9_add_archive_flag.py
Comment (dominik003) [reply backend/capellacollab/alembic/versions/f7bf9456cfc9_add_archive_flag.py:1]: Done

**Review by MoritzWeber0 — CHANGES_REQUESTED** [review]

File: backend/capellacollab/settings/modelsources/t4c/injectables.py [hunk 6]
```diff
@@ -22,3 +22,18 @@ def get_existing_instance(
             "reason": f"The t4c instance with the id {t4c_instance_id} does not exist.",
         },
     )
+
+
+def get_existing_unarchived_instance(
+    t4c_instance_id: int, db: orm.Session = fastapi.Depends(database.get_db)
+):
+    t4c_instance = get_existing_instance(t4c_instance_id, db)
```
[the remark this answers is not in the document]
Comment (dominik003) [reply backend/capellacollab/settings/modelsources/t4c/injectables.py:30]: Pipeline is now green

**Review by MoritzWeber0 — APPROVED** [review]
